cheri_compressed_cap_128.h

```diff
@@ -62,6 +92,35 @@ typedef uint64_t cc128_addr_t;
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wpedantic"
 enum {
+#ifdef IS_MORELLO
+    // Morello HW perms actually 127..116, and 111...100. But seperating the fields is just a headache, and would make
+    // other code more complex. Pretend that they are all HW Perms.
+    _CC_FIELD(HWPERMS, 127, 110),
+    _CC_FIELD(UPERMS, 111, 112),
+    // Should _CC_FIELD(UPERMS, 115, 112), if that wouldn't cause a double count because of above.
+    _CC_FIELD(OTYPE, 109, 95),
+    _CC_FIELD(EBT, 94, 64),
+// This is a bit dodgy. This enum only really works for non-address bits.
+// Just provide nonsense values that will make the length of the range 0.
+// Should really be `_CC_FIELD(FLAGS, 63, 56)', if this stuff applied to the address
+#define MORELLO_FLAG_BITS 8
+    _CC_FIELD(FLAGS, 64, 65),
```

Comment:
Does a zero-size field also work?

Reply:
This is a zero-size field. Both start and end are inclusive, so length is difference plus 1, so to get zero length end has to weirdly be one less than start.

Reply:
I clearly should not be allowed to write code...